packages/auth/auth0-react/src/layouts/Root.tsx

```diff
@@ -0,0 +1,19 @@
+import { Outlet } from "react-router-dom";
+
+import Header from "../components/Header";
+import { Auth0ProviderWithNavigate } from "../providers/auth0ProviderWithNavigate";
+
+const Root = () => {
+  return (
+    <Auth0ProviderWithNavigate>
```

Comment:
This is the most important consideration, right here. We have to wrap the Auth0 context provider around the main `<Outlet />` where pages are rendered in the router **and** write the router in a very prescriptive way.

Make sure to add the TODO to the README because there will be a lot of backtracking in the future if you don't.